src/data/api.js

```diff
@@ -31,59 +29,40 @@ async function wrappedFetch(url, body) {
   //   200's can be good { data } or bad { errors, data }
   //   422 is also possible, e.g., if the gql syntax is wrong, { errors }
   // buffalo responses more similarly mimic REST, error formats follow two patterns though:
-  //   { code, error, trace } or { code, key }
+  //   { code, error } or { code, key }
   //   the message to display will either be in { error } or it will need to be derived from { key }, i.e., 
-  //   { key } is intended for looking up the appropriate message
+  //     key is intended for looking up the appropriate message
 
-  if (contents.errors) { 
-    // must be a gql "bad" response
-    throwError(contents.errors[0].message, response.status)
-  }
-
-  // gql "bad" 200's are out of the picture now, this should be all good, we're finished
   if (response.ok) { 
     return contents
   }
 
-  // http response was not a 200 family, therefore begin dealing with the individual situations
-
   if (response.status === 401) {
-    // user is not authenticated yet... or anymore (their credentials may have expired)
     //TODO: clear user store without creating a circular dependency on user.js
-    token.reset() // "expire" their local credentials
-
-    // we need to get the user over to the login page but we may want to set up some additional
-    // handling in case they were in the middle of something in the app already, i.e., they
-    // already authenticated but their credenitals expired.
-    const currentRoute = get(location)
-    let loginRoute = '/login'
-    if (! ['/', loginRoute].includes(currentRoute)) {
-      // they were on some page in the app already and lost their credentials,
-      // we'll want to give them a little info and get them back to where they were.
-      contents.error = 'You will need to sign in first before going to that page'
-
-      loginRoute += `?return-to=${currentRoute}`      
-    }
-
-    push(loginRoute)
+    token.reset() // expire local credentials if they exist
   }
 
   // if there's a key, the message must be derived
   if (contents.key) {
     contents.error = polyglot.t(contents.key)
   }
 
-  throwError(contents.error, response.status)
+  // buffalo => `error`, gql => errors[0].message
+  throwError(contents.error || contents.errors[0].message, response.status)
 }
 
 export async function gql(query) {
   const body = JSON.stringify({
     query
   })
   
-  const { data } = await wrappedFetch('gql', body)
-   
-  return data
+  const response = await wrappedFetch('gql', body)
+  
+  if (response.errors) { 
+    throwError(contents.errors[0].message)
```

Comment:
Looks like this `contents` should be `response`:
```suggestion
    throwError(response.errors[0].message)
```

Reply:
this should be ok already